scan/scan.go

```diff
@@ -237,10 +247,43 @@ func (s *sourcescanner) walkSource(conn redis.Conn, path string, f os.FileInfo,
 		return nil, nil
 	}
 
+	var dfData DummyFile
+	dummyFile := GetConfig().DummyFiles
+
 	d := new(filedata)
 	d.path = path[len(GetConfig().Repository):]
-	d.size = f.Size()
-	d.modTime = f.ModTime()
+
+	if dummyFile {
+		raw, err := ioutil.ReadFile(path)
+		if err != nil {
+			fmt.Println(err.Error())
+			fmt.Println("Failed to read file, DELETING!")
+			os.Remove(path)
```

Comment:
This is way too dangerous and can result in a massive data loss if the redirector is also the main rsync server. Just ignore the file and display an error to the console instead.